fix_includes.py

```diff
@@ -1963,7 +1964,10 @@ def FixFileLines(iwyu_record, file_lines, flags):
   # Add a sentinel decorated move-span, to make life easy, and sort.
   decorated_move_spans.append(((len(file_lines), len(file_lines)),
                                _EOF_KIND, '', []))
-  decorated_move_spans.sort()
+  if flags.reorder:
+    decorated_move_spans.sort()
+  else:
+     decorated_move_spans.sort(key=lambda x: x[0:-2])
```

Comment:
What's the intent here? To sort by one-letter file extension (i.e. `.h`)? I don't think this holds water -- headers can be arbitrarily named, and stuff like `.hpp` and `.inl` is pretty common at least.

Reply:
Oh wait, the slice is the entire string except the trailing two chars... I still don't understand.

Reply:
Each element of `decorated_move_spans` is a tuple; the intent is to remove the key and text from each tuple so it can only sort by the span and kind.  This will preserve the order within each kind, though it only preserves the order it's already in.

Reply:
Aah, it's a tuple, I completely missed that! Please add a little comment to describe the intent, something like `# Sort only by (span, kind), not the rest of the decoration`.

Now that I understand this better -- couldn't you just take `x[0:2]` to make sure you get the first two, rather than everything but the last two?